pkg/util/docker/containers.go

```diff
@@ -348,3 +373,24 @@ func correctMissingIPs(addrs []containers.NetworkAddress, hostIPs []string) []co
 	}
 	return correctedAddrs
 }
+
+// cross IP addresses with ports
+func crossIPsWithPorts(addrs []containers.NetworkAddress, ports []nat.Port) []containers.NetworkAddress {
+	res := []containers.NetworkAddress{}
```

Comment:
nit:  this can be pre-sized  (feel free to ignore since both lists should always be quite small)